ecs-init/volumes/state_manager.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package volumes
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+const (
+	// PluginStatePath is the directory path to the plugin state information
+	PluginStatePath = "/var/lib/ecs/data/"
+	// PluginStateFile contains the state information of the plugin
+	PluginStateFile = "ecs_volume_plugin.json"
+	// PluginStateFileAbsPath is the absolute path of the plugin state file
+	PluginStateFileAbsPath = "/var/lib/ecs/data/ecs_volume_plugin.json"
+)
+
+var lock sync.Mutex
```

Comment:
having a package level lock is a little scary. We should probably bring this into  a state saver object.

Reply:
that was my initial thought too. 
two things considered:
1. the struct will only have lock as the field
2. since initially Volume plugin interface was going to be persisted as is, this object would be a global var.

But I can change this to having a different model for persistence and have a state manager object as well as part of the volume plugin.